fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/derby/DerbyResourceDAO.java

```diff
@@ -89,26 +88,13 @@ public Resource  insert(Resource resource, List<ExtractedParameterValue> paramet
         logger.entering(CLASSNAME, METHODNAME);
 
         final Connection connection = getConnection(); // do not close
-        Integer resourceTypeId;
         Timestamp lastUpdated;
-        boolean acquiredFromCache;
         long dbCallStartTime;
         double dbCallDuration;
 
         try {
-            resourceTypeId = getResourceTypeIdFromCaches(resource.getResourceType());
-            if (resourceTypeId == null) {
-                acquiredFromCache = false;
-                resourceTypeId = getOrCreateResourceType(resource.getResourceType(), connection);
-                this.addResourceTypeCacheCandidate(resource.getResourceType(), resourceTypeId);
-            } else {
-                acquiredFromCache = true;
-            }
-
-            if (logger.isLoggable(Level.FINE)) {
-                logger.fine("resourceType=" + resource.getResourceType() + "  resourceTypeId=" + resourceTypeId +
-                         "  acquiredFromCache=" + acquiredFromCache + "  tenantDatastoreCacheName=" + ResourceTypesCache.getCacheNameForTenantDatastore());
-            }
+            // check that the resource type is valid according to the database
+            getResourceTypeId(resource.getResourceType());
```

Comment:
the method name / call implies that this returns a value.  
While the comments are helpful, it's a little less readable after that point.

Reply:
Yeah, this method does return a value, but if I assign it to a variable it would introduce a warning about it not being used. The reason for making the call is just a defensive check that the resource type is valid...because the procedure may behave oddly if the resource type name isn't a known value. We currently return Integer, which means I could add an assert that the value is non-null.

Reply:
Hopefully this makes it a little more clear:
```
            Objects.requireNonNull(getResourceTypeId(resource.getResourceType()));
```